feat(Calculator): use jdk 1.8, add Interface, Test, Test resources.

sonanDev opened the pull request:
- Not finished yet.
- actually, this is a pull request test....

Commit:
feat(Calculator): use jdk 1.8, add Interface, Test, Test resources.

- Not finished yet.
- actually, this is a pull request test....

iltaek commented on src/main/java/com/sonan/message/Constant.java:
```diff
@@ -0,0 +1,11 @@
+ * @created: 2019-12-13
+ * @since: 1.8
+ */
+public class Constant {
```
상수만 따로 모아놓으면 좋을 것 같다는 생각은 좋아요 👍 
그런데 사용하는 상수가 하나 밖에 없는데 굳이 따로 클래스로 빼는 것 보다는
실제 상수를 사용하는 객체에게 관리를 위임하는 것은 어떨까요?

iltaek commented on src/main/java/com/sonan/message/Constant.java:
```diff
@@ -0,0 +1,11 @@
+ * @since: 1.8
+ */
+public class Constant {
+  public static final String nullNotAvailable = "The null is not available for calculation.";
```
상수명은 대문자로 하는 것이 컨벤션입니다.
구글 코딩 컨벤션의 [5.2.4절](https://google.github.io/styleguide/javaguide.html#s5.2.4-constant-names) 을 참고해주세요

iltaek commented on src/main/java/com/sonan/util/Calculator.java:
```diff
@@ -0,0 +1,18 @@
+package com.sonan.util;
+
+import java.util.Optional;
```
불필요한 import는 삭제해주세요

iltaek commented on src/main/java/com/sonan/util/NullNotCalculateException.java:
```diff
@@ -0,0 +1,31 @@
+ * @created: 2019-12-13
+ * @since: 1.8
+ */
+public class NullNotCalculateException extends RuntimeException {
```
상세한 예외처리를 위한 새로운 예외 클래스 도입 👍

iltaek commented on src/test/java/com/sonan/util/NumberCalculatorTest.java:
```diff
@@ -0,0 +1,48 @@
+class NumberCalculatorTest {
+  private static final Logger logger = LoggerFactory.getLogger(NumberCalculatorTest.class);
+
+  @DisplayName("Should Be number. cover with any real number.")
```
한글로 써도 돼요 ㅎㅎ

iltaek commented on src/test/java/com/sonan/util/NumberCalculatorTest.java:
```diff
@@ -0,0 +1,48 @@
+
+  @DisplayName("Should Be number. cover with any real number.")
+  @ParameterizedTest(name = "{index} => num={0}")
+  @CsvFileSource(resources = "/csv/NumberCalculatorTestInitNumber.csv")
```
굳이 파일에서 읽는 방법을 사용해야할까요?
@CsvSource 사용법을 찾아보세요!

iltaek commented on src/test/java/com/sonan/util/NumberCalculatorTest.java:
```diff
@@ -0,0 +1,48 @@
+ * @since: 1.8
+ */
+class NumberCalculatorTest {
+  private static final Logger logger = LoggerFactory.getLogger(NumberCalculatorTest.class);
```
단위테스트에 굳이 로거가 필요할까요?

iltaek reviewed (changes requested):
요즘 바쁘신데도 조금이라도 신경쓰고 진행하시려는 모습이 보기 좋습니다! 💯 
아직 미완성인 곳을 제외한 부분에 몇 가지 생각해볼만한 점을 적어놨으니 확인해보세요~! :)

Commit:
feat(Calculator): BigDecimal 지원하는 식으로 진행.

- Calculator 인터페이스 generic으로 변경.
- NumberGenerator 인터페이스 추가. number를 생성할 수 있다.
- BigDecimalCalculator, BigDeciamlGenerator add관련 구현.
